src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,196 @@
+package duke.storage;//package duke;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The Storage class manages the loading and saving of tasks to a file.
+ */
+public class Storage {
+    String filePath;
+    ArrayList<Task> tasks;
+
+    /**
+     * Constructs a Storage object with the given file path.
+     *
+     * @param filePath the file path where tasks are stored
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Validates if the start date is before the end date.
+     *
+     * @param startDate the start date
+     * @param endDate   the end date
+     * @return true if the start date is before the end date, false otherwise
+     */
+    private static boolean isValidDates(LocalDate startDate, LocalDate endDate) {
+        return startDate.isBefore(endDate);
+    }
+
+    /**
+     * Loads tasks from the file.
+     *
+     * @return the list of tasks loaded from the file
+     */
+    public ArrayList<Task> loadTasksFromFile() {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            System.out.println("File path is incorrect... Quitting...");
+            return null; // Handle the case where the file doesn't exist
+        }
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.length() < 7) {
+                    System.out.println("Error found: Bad input");
+                    System.out.println(line);
+                }
+                char taskType = line.charAt(1);
+                char isDoneX = line.charAt(4);
+                if (taskType == 'T') {
+                    String todoDescription = line.substring(7);
+                    if (isDoneX == 'X') {
+                        Task newTodo = new TodoTask(todoDescription, true);
+                        this.tasks.add(newTodo);
+                    } else if (isDoneX == ' ') {
+                        Task newTodo = new TodoTask(todoDescription);
+                        this.tasks.add(newTodo);
+                    } else {
+                        System.out.println("____________________________________________________________");
```

Comment:
Can consider keeping the line as a variable then using it here to avoid the copy and pasting of the line.